app/kuma-dp/pkg/dataplane/probes/http.go

```diff
@@ -0,0 +1,141 @@
+package probes
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	kube_core "k8s.io/api/core/v1"
+
+	"github.com/kumahq/kuma/pkg/plugins/runtime/k8s/probes"
+	"github.com/kumahq/kuma/pkg/version"
+)
+
+func (p *Prober) probeHTTP(writer http.ResponseWriter, req *http.Request) {
+	// /<port>/<original-path>
+
+	upstreamScheme := getScheme(req)
+	timeout := getTimeout(req)
+
+	transport := createHttpTransport(upstreamScheme, p.isPodAddrIPV6)
+	client := &http.Client{
+		Timeout:   timeout,
+		Transport: transport,
+	}
+
+	upstreamReq, err := buildUpstreamReq(req, upstreamScheme, p.podAddress)
+	if err != nil {
+		logger.V(1).Info("unable to create upstream request", "error", err)
+		writeProbeResult(writer, Unknown)
+		return
+	}
+
+	res, err := client.Do(upstreamReq)
+	if err != nil {
+		logger.V(1).Info("unable to request upstream server", "error", err)
+		writeProbeResult(writer, Unhealthy)
+		return
+	}
+	defer func() {
+		err = res.Body.Close()
+		if err != nil {
+			logger.V(1).Info("error closing upstream request", "error", err)
+		}
+	}()
+
+	b, err := readAtMost(res.Body, maxRespBodyLength)
+	if err != nil {
+		if errors.Is(err, errLimitReached) {
+			logger.V(1).Info("non fatal body truncation for %s, Response: %v", req.URL.String(), *res)
+		} else {
+			logger.V(1).Info("failed to read response body", "err", err)
+			writeProbeResult(writer, Unhealthy)
+			return
+		}
+	}
+
+	// from [200,400)
+	body := string(b)
+	if res.StatusCode >= http.StatusOK && res.StatusCode < http.StatusBadRequest {
+		logger.V(1).Info(fmt.Sprintf("probe succeeded for %s", upstreamReq.URL.Path),
+			"headers", upstreamReq.Header, "body", body)
+		writeProbeResult(writer, Healthy)
+		return
+	}
+
+	logger.V(1).Info(fmt.Sprintf("probe failed for %s", upstreamReq.URL.Path),
```

Comment:
nit: do we need format, or can we just add `"path", upstreamReq.URL.Path` to the log